types/react-sizes/react-sizes-tests.tsx

```diff
@@ -26,4 +25,4 @@ const TestComponent: React.ComponentType<TestProps> = ({ foo, width, height }) =
     );
 };
 
-WithSizes(mapSizesToProps)(TestComponent); // $ExpectType ComponentType<TestProps>
+WithSizes<ReturnType<typeof mapSizesToProps>, TestProps>(mapSizesToProps)(TestComponent); // $ExpectType ComponentType<Pick<TestProps, "foo">>
```

Comment:
The `$ExpectedType` breaks when `dtslint` runs it with TypeScript 4.3:

```
ERROR: 28:1  expect  TypeScript@4.3 expected type to be:
  ComponentType<Pick<TestProps, "foo">>
got:
  ComponentType<Omit<TestProps, "width" | "height">>
```

I'm suspecting that `Omit` has become a first-class utility type in 4.3, or something along those lines

Reply:
Not really sure how I should proceed. I could bump the minimum TypeScript version, but doing that just to make the tests pass doesn't feel right. `Omit` should work the same in a real world case.

Another way to solve this is how `@types/lodash` does. It [uses its own definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/dd57967bc4ffdbe6f173363c7fbbfff84b3c0f97/types/lodash/common/common.d.ts#L5), but that was 3 years ago. I believe this is because `Omit` wasn't part of TypeScript at the time.

Perhaps I should just open an issue about it in `dtslint`?

Reply:
Why don't we just use Omit here?

Reply:
We are, but it resolves to a `Pick` in earlier version of TypeScript

Reply:
Is the CI only running with 4.3? Then maybe we can just use the Omit in the test? I mean, the types still work for all versions >= 3.5, so this might be okay 🤔 Otherwise we could [follow these instructions and provide separate files](https://github.com/DefinitelyTyped/DefinitelyTyped#i-want-to-use-features-from-very-new-typescript-versions) for < 4.3 and >= 4.3. But since the types are exactly the same and only the test is affected, I don't really see the point.

Reply:
I haven't verified it on the CI, but it seems to run all the versions from 3.5 up to 4.3 locally. I can push a test commit tomorrow to see if it behaves the same here.

But yeah, I agree, feels a bit overkill to have separate files for this. Especially since the type is essentially the same

Reply:
Pushed that (ded7965c44219bbfe827490940b74af0b31c2ccb) just now, seems like the CI behaves the same as when tests are run locally:

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51460/checks?check_run_id=2069570737#step:5:80